src/util/piped_process.h

```diff
@@ -0,0 +1,39 @@
+#ifndef CPROVER_UTIL_PIPED_PROCESS_H
+#define CPROVER_UTIL_PIPED_PROCESS_H
+
+// TODO: Revisit buffer size and storage location.
+// Note that this may impact number of characters correctly communicated
+// by the pipe, to check/fix.
+#define BUFSIZE 2048
+
+class piped_processt
+{
+  enum class process_statet
+  {
+    NOT_CREATED,
+    CREATED,
+    STOPPED
+  };
+
+  char **split_command_args(const std::string &command);
```

Comment:
:hammer_and_pick: This member function doesn't appear to need the state of the class. Can this be refactored into a free function? That would make the class cleaner / smaller and make the function unit-testable in isolation.

Reply:
I'd get rid of `split_command_args` -- let the user pass in a vector of strings, i.e., using the interface of `run`.

Reply:
This function is gone. Therefore I am going to consider this  comment to be resolved.